lib/moonshot/stack.rb

```diff
@@ -176,14 +176,35 @@ def get_stack(name)
       raise "Could not describe stack: #{name}"
     end
 
+    def upload_template_to_s3
+      raise 'The S3 bucket to store the template in is not configured.' unless @config.template_s3_bucket # rubocop:disable LineLength
+
+      s3_object_key = "#{@name}-#{File.basename(template.filename)}"
+      s3_client = Aws::S3::Client.new
+      s3_client.put_object(
+        bucket: @config.template_s3_bucket,
+        key: s3_object_key,
+        body: template.body
+      )
+      template_url = "http://#{@config.template_s3_bucket}.s3.amazonaws.com/#{s3_object_key}"
+      puts "The template has been uploaded to #{template_url}"
```

Comment:
For things that happen during create/update/delete lifecycle, please use @ilog's threaded logging. Take a look at  `#delete` to get an idea of what to do.